base/ltmarks.dtx

```diff
@@ -784,40 +807,47 @@
 % \subsection{Updating mark structures}
 %
 %
-%  \begin{macro}{\l_@@_box,\g_@@_tmp_tl,\g_@@_new_top_tl}
-%    For some operations we need a temporary private box and two
+%  \begin{macro}{\l_@@_box,\l_@@_ii_box,\g_@@_tmp_tl,\g_@@_new_top_tl}
+%    For some operations we need two temporary private boxes and two
 %    private global token lists.
 %    \begin{macrocode}
 \box_new:N \l_@@_box
+\box_new:N \l_@@_ii_box
 \tl_new:N  \g_@@_tmp_tl
 \tl_new:N  \g_@@_new_top_tl
 %    \end{macrocode}
 %  \end{macro}
 %
 %
-%  \begin{macro}{\@@_update_structure:nn}
-%
-%    This function updates the mark structures. The first argument is
-%    the region to update and second argument receives the material
-%    that holds the marks. Out of this material we extract the first
-%    and last marks for all classes (if there are any) to do the
-%    assignments.
 %
+%  \begin{macro}{\@@_extract_and_handle_marks:nnn}
+%    
 %    \begin{macrocode}
-\cs_new_protected:Npn \@@_update_structure:nn #1#2
-  {
-%    \end{macrocode}
-%    First thing we do is copying the current structure to
-%    \texttt{previous-...}; this leaves the current structure
-%    untouched so we can update it class by class (which is necessary).
-%    \begin{macrocode}
-    \@@_update_structure_alias:nn { previous-#1 } {#1}
+\cs_new_protected:Npn \@@_extract_and_handle_marks:nnn #1#2#3 {
 %    \end{macrocode}
-%    Getting the first and last marks out of the material in \verb=#2=
+%    This macro expects a region name as its third argument and
+%    vertical material (not boxed) as its fourth. It attempts to
+%    extract mark information from \verb/#4/ and if it succeeds It
+%    executes \verb/#1{#3}/ to do something with the marks, e.g., to
+%    update the mark structure for the region.
+%  
+%    If it finds infinite negative glue it generates an error message
```

Comment:
Interesting thought, but I don't think there is a way to detect then that the error has happened and it is an error, ie the user is losing the marks and so I think they should learn that this is the case. But perhaps if it is enough if that is something you can find in the .log? It would certainly has the advantage that then the processing doesn't stop if they can't avoid the infinite negative glue. Maybe.

Reply:
hmm, back to the drawing board perhaps. If we run with \interactionmode 0, the error only appears in the log and we actually get to the marks even if there is real infinite negative glue (because TeX makes it finite).

In other words, if we accept that we may see
```
! Infinite glue shrinkage found in box being split.
\__box_dim_eval:n ...box_dim_eval:w #1\scan_stop: 
l. ...   }
The box you are \vsplitting contains some infinitely
shrinkable glue, e.g., `\vss' or `\vskip 0pt minus 1fil'.
Such glue doesn't belong there; but you can safely proceed,
since the offensive shrinkability has been made finite.
! Infinite glue shrinkage found in box being split.
```
in the log then we can get at the marks in all cases, which seems better then issuing an error or a warning that it doesn't work. 

If we want to we could additionally do
```
\wlog{Any immediately following 'Infinite glue shrinkage' error is harmless and can be ignored}
```
but I'm not sure this is better, because most of the time there will be none and we only end up with these extra lines. Thoughts?

Reply:
You could use `\ignore_infinite_shrinkage_error  defined as `\let\ignore_infinite_shrinkage_error\__box_dim_eval:n ` perhaps....

Reply:
```

\expandafter\def\csname Ignore any infinite glue shrinkage error on this line\endcsname{
\setbox0\vbox{aaa \vss bbb}%
\setbox2=\vsplit0 to \maxdimen
}
\expandafter\def\expandafter\foo\expandafter{%
  \csname Ignore any infinite glue shrinkage error on this line\endcsname}
{
\interactionmode=0
\foo
}
```

makes a log

```
! Infinite glue shrinkage found in box being split.
\Ignore any infinite glue shrinkage error on this line ...
                                                  
l.13 \foo
```